### PR TITLE
doc: update INSTALL.md about opam and platform versions

### PR DESCRIPTION
**Pull request by Alizter**
I've added some more information about the Coq Platform and opam versions as discussed in #2045.

### CODE REVIEW

**Review by jdchristensen — COMMENTED** [review]
Looks fine to me, but I haven't tested the different  install methods.

**Review by mikeshulman — COMMENTED** [review]
Looks good, thanks.  If the development versions in opam still work, can we add something about how to use them?

**Comment by Alizter** [comment]
@mikeshulman I checked myself, and we need the `coq.dev` version of `coq`. This is found in
```
$ opam repo add coq-core-dev https://coq.inria.fr/opam/core-dev
```
so I've added that to `INSTALL.md`. I also added a note about pinning Coq itself, as Coq master may have issues and we don't want constant rebuilds.

**Comment by Alizter** [comment]
@mikeshulman Any updates on this?

File: INSTALL.md [hunk 1]
```diff
@@ -66,12 +79,23 @@ library is `coq-hott` inside the coq-archive.
...
 We also have the current development versions of the library available via
 `opam`. For this however, you will need to add the dev coq-archive packages:
 ```shell
+$ opam repo add coq-core-dev https://coq.inria.fr/opam/core-dev
 $ opam repo add coq-extra-dev https://coq.inria.fr/opam/extra-dev
 ```
 
+This will make `coq.dev` the latest available version of `coq`. You can pin
+`coq` to a stable version by running `opam pin add coq.dev 8.19.1` for example.
+
```
Comment (mikeshulman) [inline INSTALL.md:98]:
```suggestion
Then install the library with `opam install coq-hott`, as for the released version.

```

**Review by mikeshulman — APPROVED** [review]

**Pull request merged**